pkg/storage/cacher_test.go

```diff
@@ -79,79 +73,44 @@ func TestList(t *testing.T) {
 	podFooPrime := makeTestPod("foo")
 	podFooPrime.Spec.NodeName = "fakeNode"
 
-	testCases := []*etcd.Response{
-		{
-			Action: "create",
-			Node: &etcd.Node{
-				Value:         string(runtime.EncodeOrDie(testapi.Default.Codec(), podFoo)),
-				CreatedIndex:  1,
-				ModifiedIndex: 1,
-			},
-		},
-		{
-			Action: "create",
-			Node: &etcd.Node{
-				Value:         string(runtime.EncodeOrDie(testapi.Default.Codec(), podBar)),
-				CreatedIndex:  2,
-				ModifiedIndex: 2,
-			},
-		},
-		{
-			Action: "create",
-			Node: &etcd.Node{
-				Value:         string(runtime.EncodeOrDie(testapi.Default.Codec(), podBaz)),
-				CreatedIndex:  3,
-				ModifiedIndex: 3,
-			},
-		},
-		{
-			Action: "set",
-			Node: &etcd.Node{
-				Value:         string(runtime.EncodeOrDie(testapi.Default.Codec(), podFooPrime)),
-				CreatedIndex:  1,
-				ModifiedIndex: 4,
-			},
-			PrevNode: &etcd.Node{
-				Value:         string(runtime.EncodeOrDie(testapi.Default.Codec(), podFoo)),
-				CreatedIndex:  1,
-				ModifiedIndex: 1,
-			},
-		},
-		{
-			Action: "delete",
-			Node: &etcd.Node{
-				CreatedIndex:  1,
-				ModifiedIndex: 5,
-			},
-			PrevNode: &etcd.Node{
-				Value:         string(runtime.EncodeOrDie(testapi.Default.Codec(), podBar)),
-				CreatedIndex:  1,
-				ModifiedIndex: 1,
-			},
-		},
+	if err := etcdStorage.Create(context.TODO(), etcdtest.AddPrefix("pods/ns/foo"), podFoo, nil, 0); err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if err := etcdStorage.Create(context.TODO(), etcdtest.AddPrefix("pods/ns/bar"), podBar, nil, 0); err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	if err := etcdStorage.Create(context.TODO(), etcdtest.AddPrefix("pods/ns/baz"), podBaz, nil, 0); err != nil {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
-	// Propagate some data to etcd.
-	for _, test := range testCases {
-		fakeClient.WatchResponse <- test
+	// To force "update" behavior of Set(), we need to set ResourceVersion of previous version of object.
+	podFooPrime.ObjectMeta.ResourceVersion = "3"
+	if err := etcdStorage.Set(context.TODO(), etcdtest.AddPrefix("pods/ns/foo"), podFooPrime, nil, 0); err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	podFooPrime.ObjectMeta.ResourceVersion = ""
+
+	deleted := api.Pod{}
+	if err := etcdStorage.Delete(context.TODO(), etcdtest.AddPrefix("pods/ns/bar"), &deleted); err != nil {
+		t.Errorf("Unexpected error: %v", err)
 	}
 
 	result := &api.PodList{}
-	if err := cacher.List(context.TODO(), "pods/ns", 5, storage.Everything, result); err != nil {
-		t.Errorf("unexpected error: %v", err)
+	if err := cacher.List(context.TODO(), "pods/ns", 8, storage.Everything, result); err != nil {
```

Comment:
same comment as above, otherwise folks reading would need to follow "why" it's 8.